app/models/setting/content.rb

```diff
@@ -143,6 +143,9 @@ def self.default_settings
       self.set('host_profile_assume', N_("Allow new Host registrations to assume registered profiles with matching hostname " \
                   "as long as the registering DMI UUID is not used by another host."),
                true, N_('Host Profile Assume')),
+      self.set('host_profile_assume_build_can_change', N_("Allow Host registrations to bypass 'Host Profile Assume' " \
```

Comment:
@jcpunk I think there's a discrepancy between this description and the actual behavior.

The wording states that the new setting supercedes `host_profile_assume` if the host is in build mode ie they are not coupled. The implementation states that `if host_profile_assume is true and host_profile_assume_build_can_change is true and the host is in build mode, then the DMI UUID may change when registering` ie they are coupled.

if the setting description is correct then I think the logic of `dmi_uuid_change_allowed?` would be more like this:

```
      def dmi_uuid_change_allowed?(host, host_uuid_overridden)                                                                                                                                
        if host_uuid_overridden                                                                                                                                                               
          true                                                                                                                                                                                
        elsif host.build && Setting[:host_profile_assume_build_can_change]                                                                                                                    
          true                                                                                                                                                                                
        else                                                                                                                                                                                  
          Setting[:host_profile_assume]                                                                                                                                                       
        end                                                                                                                                                                                   
      end 
```

In other words host_uuid_overridden > host.build & host_profile_assume_build_can_change > host_profile_assume

Reply:
Yeah, that does look better.

I'll have a new patch up shortly.